ipn/ipnlocal/network-lock.go

```diff
@@ -430,6 +430,45 @@ func (b *LocalBackend) NetworkLockKeyTrustedForTest(keyID tkatype.KeyID) bool {
 	return b.tka.authority.KeyTrusted(keyID)
 }
 
+// NetworkLockSign signs the given node-key and submits it to the control plane.
+// rotationPublic, if specified, must be an ed25519 public key.
+func (b *LocalBackend) NetworkLockSign(nodeKey key.NodePublic, rotationPublic []byte) error {
+	b.mu.Lock()
```

Comment:
Maybe move everything that needs the lock into a helper function so you can defer the unlock?

Reply:
Re-did everything into a closure and its much cleaner, thanks.